cpp/tests/test_abm_world.cpp

```diff
@@ -496,65 +496,211 @@ TEST(TestWorld, checkParameterConstraints)
+    p_no_test.set_compliance(mio::abm::InterventionType::Testing, 0.4);
+    p_no_isolation.set_compliance(mio::abm::InterventionType::Isolation, 0.4);
+
+    world.evolve(t, dt);
+
+    // The complied person is allowed to be at work and wear the required mask
+    EXPECT_EQ(p_compliant_go_to_work.get_location(), work_id);
+    EXPECT_EQ(p_compliant_go_to_work.get_mask().get_type(), mio::abm::MaskType::FFP2);
+
+    // The complied person is allowed to be at school and don't wear mask
+    EXPECT_EQ(p_compliant_go_to_school.get_location(), school_id);
+    EXPECT_EQ(p_compliant_go_to_school.get_mask().get_type(), mio::abm::MaskType::None);
+
+    // The person, who does not wear mask, is not allowed to be in location
+    EXPECT_EQ(p_no_mask.get_mask().get_type(), mio::abm::MaskType::None);
+    EXPECT_NE(p_no_mask.get_location(), work_id);
+
+    // The person, who does not want test, is not allowed to be in location
+    EXPECT_NE(p_no_test.get_location(), work_id);
+
+    // The person does not want to isolate when the test is positive
+    EXPECT_FALSE(p_no_isolation.is_in_quarantine(t, world.parameters));
+
+    // Using trip list
+    mio::abm::TripList& trip_list = world.get_trip_list();
+    mio::abm::Trip trip1(p_compliant_go_to_work.get_id(), t, work_id, home_id);
+    mio::abm::Trip trip2(p_compliant_go_to_school.get_id(), t, school_id, home_id);
+    mio::abm::Trip trip3(p_no_mask.get_id(), t, work_id, home_id);
+    mio::abm::Trip trip4(p_no_test.get_id(), t, work_id, home_id);
+    mio::abm::Trip trip5(p_no_isolation.get_id(), t, work_id, home_id);
+    trip_list.add_trip(trip1);
+    trip_list.add_trip(trip2);
+    trip_list.add_trip(trip3);
+    trip_list.add_trip(trip4);
+    trip_list.add_trip(trip5);
+    world.migrate(p_compliant_go_to_work.get_id(), home_id); // reset person location
+    world.migrate(p_compliant_go_to_school.get_id(), home_id); // reset person location
+    world.use_migration_rules(false);
+    world.evolve(t, dt);
+
+    // The complied person is allowed to be at work and wear the required mask
+    EXPECT_EQ(p_compliant_go_to_work.get_location(), work_id);
+    EXPECT_EQ(p_compliant_go_to_work.get_mask().get_type(), mio::abm::MaskType::FFP2);
+
+    // The complied person is allowed to be at school and don't wear mask
+    EXPECT_EQ(p_compliant_go_to_school.get_location(), school_id);
+    EXPECT_EQ(p_compliant_go_to_school.get_mask().get_type(), mio::abm::MaskType::None);
+
+    // The person, who does not wear mask, is not allowed to be in location
+    EXPECT_EQ(p_no_mask.get_mask().get_type(), mio::abm::MaskType::None);
+    EXPECT_NE(p_no_mask.get_location(), work_id);
+
+    // The person, who does not want test, is not allowed to be in location
+    EXPECT_NE(p_no_test.get_location(), work_id);
+
+    // The person does not want to isolate when the test is positive
+    EXPECT_FALSE(p_no_isolation.is_in_quarantine(t, world.parameters));
```

Comment:
This strongly looks like a different test case, and probably should get its own test.
Or at least write an explanation over each section what exactly the test cases are.

Reply:
Ok, I now moved this section to another test.